xarray/indexes/multipandasindex.py

```diff
@@ -0,0 +1,175 @@
+            else:
+                new_indexes[k] = idx
+
+        #
+        # How should we deal with dropped index(es) (scalar selection)?
+        # - drop the whole index?
+        # - always return a MultiPandasIndex with remaining index(es)?
+        # - return either a MultiPandasIndex or a PandasIndex?
+        #
+
+        if not len(new_indexes):
+            return None
+        elif len(new_indexes) == 1:
+            return next(iter(new_indexes.values()))
+        else:
+            return type(self)(new_indexes)
+
+    def sel(self, labels: dict[Any, Any], **kwargs) -> IndexSelResult:
+        results: list[IndexSelResult] = []
+
+        for k, idx in self.indexes.items():
+            if k in labels:
+                results.append(idx.sel({k: labels[k]}, **kwargs))
+
+        return merge_sel_results(results)
+
+    def _get_unmatched_names(
+        self: T_MultiPandasIndex, other: T_MultiPandasIndex
+    ) -> set:
+        return set(self.indexes).symmetric_difference(other.indexes)
+
+    def equals(self: T_MultiPandasIndex, other: T_MultiPandasIndex) -> bool:
+        # We probably don't need to check for matching coordinate names
+        # as this is already done during alignment when finding matching indexes.
+        # This may change in the future, though.
+        # see https://github.com/pydata/xarray/issues/7002
+        if self._get_unmatched_names(other):
+            return False
+        else:
+            return all(
+                [idx.equals(other.indexes[k]) for k, idx in self.indexes.items()]
+            )
+
+    def join(
+        self: T_MultiPandasIndex, other: T_MultiPandasIndex, how: str = "inner"
+    ) -> T_MultiPandasIndex:
+        new_indexes = {}
+
+        for k, idx in self.indexes.items():
+            new_indexes[k] = idx.join(other.indexes[k], how=how)
+
+        return type(self)(new_indexes)
+
+    def reindex_like(
+        self: T_MultiPandasIndex, other: T_MultiPandasIndex
+    ) -> dict[Hashable, Any]:
+        dim_indexers = {}
+
+        for k, idx in self.indexes.items():
+            dim_indexers.update(idx.reindex_like(other.indexes[k]))
+
+        return dim_indexers
+
+    def roll(self: T_MultiPandasIndex, shifts: Mapping[Any, int]) -> T_MultiPandasIndex:
+        new_indexes = {}
+
+        for k, idx in self.indexes.items():
+            if k in shifts:
+                new_indexes[k] = idx.roll({k: shifts[k]})
+            else:
+                new_indexes[k] = idx
+
+        return type(self)(new_indexes)
+
+    def rename(
+        self: T_MultiPandasIndex,
+        name_dict: Mapping[Any, Hashable],
+        dims_dict: Mapping[Any, Hashable],
+    ) -> T_MultiPandasIndex:
+        new_indexes = {}
+
+        for k, idx in self.indexes.items():
+            new_indexes[k] = idx.rename(name_dict, dims_dict)
+
+        return type(self)(new_indexes)
+
+    def copy(self: T_MultiPandasIndex, deep: bool = True) -> T_MultiPandasIndex:
```

Comment:
Might as well implement the correct deep-copy behavior which passes the memo dict.